File: Java/SherlockAndSquares.java
import java.io.*;
import java.util.*;

import static java.lang.Math.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner in = new Scanner(System.in);
        
        int tests = in.nextInt();
        int down, up;
        
        for (int t = 0; t < tests; t++) {
            
            down = (int) ceil(sqrt(in.nextInt()));
            up = (int) floor(sqrt(in.nextInt()));
            
            System.out.println(up + 1 - down);
        }
    }
}

File: Ruby/SherlockAndSquares.rb

# Enter your code here. Read input from STDIN. Print output to STDOUT
t = gets.to_i
for x in 1..t do
    a,b = gets.split(' ').map(&:to_i)
    puts Math.sqrt(b).to_i - Math.sqrt(a).ceil + 1
end
